tools/shared.py

```diff
@@ -226,7 +226,14 @@ def new(*args):
     config_file = config_file.replace('\'{{{ EMSCRIPTEN_ROOT }}}\'', repr(__rootpath__))
     llvm_root = os.path.dirname(find_executable('llvm-dis') or '/usr/bin/llvm-dis')
     config_file = config_file.replace('\'{{{ LLVM_ROOT }}}\'', repr(llvm_root))
-    binaryen_root = os.path.dirname(find_executable('asm2wasm') or '') # if we don't find it, we'll use the port
+    # binaryen is a little different: it's root is not the dir containing the
+    # executable, but the dir of binaryen itself, as we need access to other things
+    # in it (like scripts/).
+    binaryen_root = os.path.dirname(os.path.dirname(find_executable('asm2wasm') or '')) # if we don't find it, we'll use the port
+    if binaryen_root:
+      # check if it in fact contains all of binaryen
+      if not os.path.exists(os.path.join('binaryen_root', 'scripts')):
```

Comment:
oops, yeah, that's definitely wrong.

this should really have a test, which would have caught that, i'll get back to this with that when i can.